scripts/contiv-network-report.sh

```diff
@@ -205,18 +202,18 @@ do
         then
             IF_NAME=$( trim "${IF_NAME_STATUS[0]}" )
             IF_NAMES+=("$IF_NAME")
-        elif [ $( trim "${IF_NAME_STATUS[0]}" ) == "L3" ]
+        elif [ "$( trim "${IF_NAME_STATUS[0]}" )" == "L3" ]
         then
             IF_IP["$IF_NAME"]=$( trim "${IF_NAME_STATUS[1]}" )
         fi
     done
 
     # Get MAC addresses for all interfaces that have an IP address
-    VPP_MAC_ADDR=$( cat "$nn"/"$VPP_MAC_ADDR_FILE" | grep -v "Name" ) || true
+    VPP_MAC_ADDR=$( grep < "$nn"/"$VPP_MAC_ADDR_FILE" -v "Name" ) || true
```

Comment:
Redirect should be after grep args.  Also grep expression should have delimiters or be full line match.